File: python/minmax.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  minmax.py
import random 

def losuj(ile, lmin, lmax):
    lista = []
    for i in range (ile):
        lista.append(random.randint(lmin, lmax))
    print (lista)
    return(lista)
    
def minmax():
    ile = int(input("ile liczb podasz? "))
    min = max = int(input("podaj liczbę: "))
    for i in range(ile - 1):
        liczba = int(input("podaj liczbę: "))
        if min > liczba:
            min = liczba
        if max < liczba:
            max = liczba
    return min, max
    
    
def minimum(liczby):
    najm = liczby[0]
    for i in liczby:
        if najm > i:
            najm = i
    return najm 
    
    
    def maximum(liczby):
    najw = liczby[0]
    for i in liczby:
        if najw < i:
            najm = i
    return najw 
    
def main(args):
    # ~min, max = minmax()
    # ~print("Najmniejsza:",min)
    # ~print("Największa:",max)
    losuj (20, 0, 50)
    print('Najmniejsza: ', minimum(liczby))
    print('Największa: ', (liczby))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/minmax.cxx
/*
 * minmax.cxx
 * 
 */


#include <iostream>

using namespace std;

void wypelnij(int tab[], int roz) {
    cout << "wprowadź " << roz << " liczb:"<< endl; 
    for(int i=0; i <roz; i++){
        cin >> tab[i];
    }
}


void drukuj(int tab[], int roz) { 
    for(int i = 0; i <roz; i++){
        cout << tab[i] << " ";
    }
}


void minmax(int tab[], int roz) {
    int minimum = tab[0];
    int maksimum = tab[0];
    for(int i = 0; i <roz; i++){
        if (minimum > tab[i])
            minimum = tab[i];
        if (maksimum < tab[i])
            maksimum = tab[i];
    }
    cout << "min. " << minimum << endl;
    cout << "max. " << maksimum << endl;
}
    

    
    
int main(int argc, char **argv)
{   
    const int rozmiar = 10;
    int tab[rozmiar];
    wypelnij(tab, rozmiar);
    drukuj(tab, rozmiar);
	return 0;
}

